test.cpp

```diff
@@ -238,6 +238,21 @@ TEST(HostnameTest, LowercasesHostname)
     EXPECT_EQ("www.testing.com", Url::Url("http://www.testing.com/FOO").host());
 }
 
+TEST(QueryTest, SanitizesQuery)
+{
+    EXPECT_EQ("a=1&b=2"    , Url::Url("http://foo.com/?a=1&&&&&&b=2"   ).query());
+    EXPECT_EQ("foo=2"      , Url::Url("http://foo.com/????foo=2"       ).query());
+    EXPECT_EQ("foo=2"      , Url::Url("http://foo.com/?foo=2&&&"       ).query());
+}
```

Comment:
I'll check to see what `url-py` does -- these tests are copied directly from there.